src/users/users.service.ts

```diff
@@ -0,0 +1,36 @@
+import { Inject, Injectable } from '@nestjs/common';
+import { UserDto } from 'src/models/user.dto';
+import { User } from 'src/models/user.entity';
+import { v4 as uuidv4 } from 'uuid';
+
+@Injectable()
+export class UsersService {
+  constructor(
+    @Inject('USERS_REPOSITORY') private usersRepository: typeof User,
+  ) {}
+
+  public async createUser(userDto: UserDto): Promise<User> {
+    return this.usersRepository.create({ id: uuidv4(), ...userDto });
+  }
+
+  public async getAllUsers(substring = '', limit = 10): Promise<User[]> {
+    return this.usersRepository.findAll<User>().then((users): User[] =>
+      users
+        .filter((user: User): boolean => user.login.includes(substring))
+        .sort((a: User, b: User): number => (a.login > b.login ? 1 : -1))
+        .slice(0, limit),
+    );
+  }
```

Comment:
It is not required, but you can use a request with sequelize options. for example:
```async findAll(loginSubstring: string, limit: number) {
    const allUsers = await this.userModel.findAll({
      where: { login: { [Op.substring]: loginSubstring }, isDeleted: false },
      order: [['login', 'ASC']],
      limit: limit,
    });
    return allUsers;
  }```